torch/csrc/jit/codegen/cuda/fusion_executor.h

```diff
@@ -0,0 +1,183 @@
+#pragma once
+#include <torch/csrc/jit/codegen/cuda/fusion.h>
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/ir_cloner.h>
+#include <torch/csrc/jit/codegen/cuda/ir_printer.h>
+#include <torch/csrc/jit/codegen/cuda/kernel.h>
+#include <torch/csrc/jit/codegen/cuda/lower2device.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+class TORCH_CUDA_API LaunchParams {
+ public:
+  unsigned int smem() const {
+    return smem_;
+  }
+  unsigned int nBlocks() const {
+    return gdimx_ * gdimy_ * gdimz_;
+  }
+
+  unsigned int nThreads() const {
+    return bdimx_ * bdimy_ * bdimz_;
+  }
+
+  unsigned int bdimx() const {
+    return (unsigned int)bdimx_ == -1 ? 1 : bdimx_;
+  }
+
+  unsigned int gdimx() const {
+    return (unsigned int)gdimx_ == -1 ? 1 : gdimx_;
+  }
+
+  unsigned int bdimy() const {
+    return (unsigned int)bdimy_ == -1 ? 1 : bdimy_;
+  }
+
+  unsigned int gdimy() const {
+    return (unsigned int)gdimy_ == -1 ? 1 : gdimy_;
+  }
+
+  unsigned int bdimz() const {
+    return (unsigned int)bdimz_ == -1 ? 1 : bdimz_;
+  }
+
+  unsigned int gdimz() const {
+    return (unsigned int)gdimz_ == -1 ? 1 : gdimz_;
+  }
+
+  void checkAndSet(
+      const int64_t incoming_val,
+      int& class_val,
+      std::string val) {
```

Comment:
`val` is not used?

Reply:
Thanks, meant to include it in the error message. Fixed.